automated install/basic-install.sh

```diff
@@ -88,16 +88,27 @@ if command -v apt-get &> /dev/null; then
   DNSMASQ_USER="dnsmasq"
 
   test_dpkg_lock() {
-    i=0
-    while fuser /var/lib/dpkg/lock >/dev/null 2>&1 ; do
-      sleep 0.5
-      ((i=i+1))
+    # Account for cloud init type installs that autorun apt on boot
+    local maxtries
+    local retVal
+    maxtries=75
+
+    for i in $(seq 1 $maxtries);
+    do
+      "${UPDATE_PKG_CACHE}"
+      retVal=$?
+      # if we find lock condition
+      if [[ $retVal -eq 100 ]]; then
+        sleep 5
+      elif [[ $retVal -eq 0 ]]; then
+        return 0
+      fi
     done
-    # Always return success, since we only return if there is no
-    # lock (anymore)
-    return 0
+    # Counter timed out, and we didn't return already, error
+    return 1
```

Comment:
We can push out the time, I've found on Vultr that it fires after one loop anyways, (apparently it only runs an update on boot, and waits till stable before actually doing any security upgrades. Otherwise boot times would be 5 minutes to spin up an instance.) I just don't want to get into a race where we're locking and then `unattended-updates|cloud-init` is updating and we're playing pingpong with the lockfile.

Thats also why I only call the lock test on the first `apt update` and dependency install, there shouldn't be a lock condition on the `PIHOLE_DEPS` installer.

So, sure, timout to 10 minutes, but just not infinite. It's a pain to SSH in to a second term and try to chase down the first shell to `kill -9` it.